File: Assets/Scripts/Framework/Quality/QualityService.cs
/********************************************************************
	created:	2018-11-9
	author:		jordenwu
	
	purpose:    质量管理
*********************************************************************/
using UnityEngine;
using JW.Common;
using JW.Framework.Asset;
using UnityEngine.Rendering;

namespace JW.Framework.Quality
{
    public class QualityService : Singleton<QualityService>
    {
        private BaseAsset _curRefectionCubMapAsset;


        public override bool Initialize()
        {
           
            return true;
        }

        public override void Uninitialize()
        {

            if (_curRefectionCubMapAsset != null)
            {
                AssetService.GetInstance().Unload(_curRefectionCubMapAsset);
                _curRefectionCubMapAsset = null;
            }
        }


        public void LoadGlobalReflectionCubeMap(float intensity, string resPath)
        {
            if (string.IsNullOrEmpty(resPath))
            {
                JW.Common.Log.LogE("LoadGlobalReflectionCubeMap Error CubeMap");
            }

            RenderSettings.reflectionIntensity = intensity;
            RenderSettings.defaultReflectionMode = DefaultReflectionMode.Custom;

            BaseAsset ba = AssetService.GetInstance().LoadPrimitiveAsset(resPath,4);
            if (ba == null)
            {
                JW.Common.Log.LogE("LoadGlobalReflectionCubeMap Error CubeMap");
            }
            else
            {
                if (_curRefectionCubMapAsset != null)
                {
                    AssetService.GetInstance().Unload(_curRefectionCubMapAsset);
                    _curRefectionCubMapAsset = null;
                }

                RenderSettings.customReflection = ba.Resource.Content as Cubemap;
                _curRefectionCubMapAsset = ba;
            }
           
        }

        public void UnloadGlobalReflectionCubeMap()
        {
            RenderSettings.customReflection = null;
            if (_curRefectionCubMapAsset!=null)
            {
                AssetService.GetInstance().Unload(_curRefectionCubMapAsset);
                _curRefectionCubMapAsset = null;
            }
        }


    }
}

File: Assets/Resources/LuaScripts/Framework/Quality/QualityService.lua
--[[********************************************************************
*		作者： jordenwu
*		时间： 2018-11-09
*		描述： 全局质量管理 光照管理
*********************************************************************--]]
local QualityServiceClass = DeclareClass("QualityServiceClass")

function QualityServiceClass:ctor()
	
	self.csS=Interaction.QualityService
	
end

function QualityServiceClass:LoadGlobalReflectionCubeMap( intensity,cubemap)
    if self.csS then
        self.csS:LoadGlobalReflectionCubeMap(intensity,cubemap)
    end
end

function QualityServiceClass:UnloadGlobalReflectionCubeMap()
    if self.csS then
        self.csS:UnloadGlobalReflectionCubeMap()
    end
end

QualityService=QualityServiceClass.new()